.pre-commit-config.yaml

```diff
@@ -0,0 +1,36 @@
+# See https://pre-commit.com for more information
+
+default_language_version:
+  python: python3.9
+
+default_stages: [commit, push]
+exclude: ^debug/
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-case-conflict
+      - id: name-tests-test
+
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.10.0
+    hooks:
+      - id: isort
+        args: ["--check"]
+
+  - repo: https://github.com/ambv/black
+    rev: 21.10b0
+    hooks:
+      - id: black
+        args: ["--check"]
+
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.910
+    hooks:
+      - id: mypy
```

Comment:
Shouldn't there be a `--ignore-missing-imports` in the args? can't test right now though

Reply:
No need since we already specified it in `pyproject.toml`